infer/src/IR/Tenv.ml

```diff
@@ -36,7 +41,10 @@ let mk_struct tenv ?default ?fields ?statics ?methods ?exported_objc_methods ?su
         Struct.internal_mk_struct ?default ?fields ?statics ?methods ?exported_objc_methods ?supers
           ?objc_protocols ?annots ?class_info ?dummy ?source_file name
       in
-      TypenameHash.replace tenv name struct_typ ;
+      (* Only replace the entry if it actually changed to reduce churn *)
+      ( match TypenameHash.find_opt tenv name with
```

Comment:
The impact of this optimisation is questionnable.
- Now we have 2 accesses instead of one
- It is rare clients will run twice `mk_struct` on the same key